protocol/consistencychecks.go

```diff
@@ -215,6 +256,45 @@ func (cc *ConsistencyChecks) verifyKeyLookup(msg *Response,
 	return CheckPassed
 }
 
+func (cc *ConsistencyChecks) verifyMonitoring(msg *Response,
+	uname string, key []byte) error {
+	if msg.Error != ReqSuccess {
+		return ErrMalformedDirectoryMessage
+	}
+	dfs := msg.DirectoryResponse.(*DirectoryProofs)
+
+	str0 := dfs.STR[0]
+	ap0 := dfs.AP[0]
+	regEp, ok := cc.RegEpoch[uname]
+
+	switch {
+	case !ok && str0.Epoch == 0 && ap0.ProofType() == m.ProofOfAbsence: /* prior history verification */
+	case ok && str0.Epoch == regEp && ap0.ProofType() == m.ProofOfAbsence: /* registration epoch */
+	case ok && str0.Epoch >= regEp+1 && ap0.ProofType() == m.ProofOfInclusion: /* after registration */
```

Comment:
Maybe define `wasUnameAbsent := ap0.ProofType() == m.ProofOfAbsence` before the `switch` and use this var in this 3 cases (with a `!` for last case)? Makes it easier to read (at least I think so; might be wrong)